gradio/processing_utils.py

```diff
@@ -284,23 +297,35 @@ def check_public_url(url: str):
     except socket.gaierror:
         raise httpx.RequestError(f"Cannot resolve hostname: {hostname}") from None
 
+    is_public = False
     for family, _, _, _, sockaddr in addrinfo:
         ip = sockaddr[0]
         if family == socket.AF_INET6:
             ip = ip.split("%")[0]  # Remove scope ID if present
 
-        if not ipaddress.ip_address(ip).is_global:
-            raise httpx.RequestError(
-                f"Non-public IP address found: {ip} for URL: {url}"
+        if ipaddress.ip_address(ip).is_global:
+            is_public = True
+            break
+
+    if is_public:
+        return url
+
+    google_resolved_ip = resolve_with_google_dns(hostname)
+    if google_resolved_ip and ipaddress.ip_address(google_resolved_ip).is_global:
+        new_parsed = parsed_url._replace(netloc=google_resolved_ip)
+        if parsed_url.port:
+            new_parsed = new_parsed._replace(
+                netloc=f"{google_resolved_ip}:{parsed_url.port}"
             )
+        return urlunparse(new_parsed)
```

Comment:
If we've verified that the url maps to a public domain, why can't we use the original url?

Reply:
https://huggingface.slack.com/archives/C06V0AAK7QB/p1724065700937259?thread_ts=1723778316.424279&cid=C06V0AAK7QB